### PR TITLE
Mustache functions

### PR DESCRIPTION
**Pull request by kanika1894**

### CODE REVIEW

File: src/configure/helper/mustacheHelper.ts [hunk 1]
```diff
@@ -127,6 +127,76 @@ export class MustacheHelper {
...
+
+            /*
+            * Checks if a string begins with some string
+            * Usage: {{#beginsWith}} StringToCheck BeginningPart true/false(for case-sensitivity) IfTrueValue IfFalseValue{{/beginsWith}}
+            */
+            "beginsWith": function () {
+                return function (text: string, render: any) {
+                    var renderedText: string = render(text);
+                    var parts = MustacheHelper.getParts(renderedText);
+                    if (parts.length < 4) {
```
Comment (bishal-pdMSFT) [inline src/configure/helper/mustacheHelper.ts:140]: shouldn't it be `5`?

Comment (bishal-pdMSFT) [reply src/configure/helper/mustacheHelper.ts:140]: Ohh okay it seems IfFalseValue is optional. Can you say so in the usage in comments

File: src/configure/helper/mustacheHelper.ts [hunk 2]
```diff
@@ -127,6 +127,76 @@ export class MustacheHelper {
...
+                    }
+
+                    var order = parts[1].toLowerCase();
+                    var arr = render(parts[0]).split(",");
+                    var sorter = 1;
+                    if (order === "dsc") {
+                        sorter = -1;
+                    }
+                    arr = arr.sort((a, b) => {
+                        if (a > b) { return 1 * sorter; }
```
Comment (bishal-pdMSFT) [inline src/configure/helper/mustacheHelper.ts:195]: should it not return true/false?

File: src/configure/helper/mustacheHelper.ts [hunk 3]
```diff
@@ -127,6 +127,76 @@ export class MustacheHelper {
...
+                    }
+                    return "{{ " + parts[0] + " }}";
+                };
+            },
+
+            /*
+            * Sort an integer array
+            * Usage: {{#intSorter}} IntegerArrayToBeSorted asc/dsc(order in which it is to be sorted) {{/intSorter}}
+            */
+            "intSorter": function () {
```
Comment (bishal-pdMSFT) [inline src/configure/helper/mustacheHelper.ts:181]: This anyways is subject to change. right?

File: src/configure/helper/mustacheHelper.ts [hunk 4]
```diff
@@ -127,6 +127,76 @@ export class MustacheHelper {
...
+                        return "";
+                    }
+                };
+            },
+
+            /*
+            * If some variable is the environment variable, adding {{ }}
+            * Usage: {{#EnvironmentVariable}} VariableName {{/EnvironmentVariable}}
+            */
+            "EnvironmentVariable": function () {
```
Comment (anuragc617) [inline src/configure/helper/mustacheHelper.ts:166]: 'EnvironmentVariable' should start with lowercase 'e' as we are using it as environmentVar in our workflow files.

File: src/configure/helper/mustacheHelper.ts (hunk 1 above)
Comment (kanika1894) [reply src/configure/helper/mustacheHelper.ts:140]: updated

File: src/configure/helper/mustacheHelper.ts (hunk 3 above)
Comment (kanika1894) [reply src/configure/helper/mustacheHelper.ts:181]: yes
As of now, it's just taking an integer array as an input
Later will include the Json array handling

File: src/configure/helper/mustacheHelper.ts (hunk 4 above)
Comment (kanika1894) [reply src/configure/helper/mustacheHelper.ts:166]: cool, updated

File: src/configure/helper/mustacheHelper.ts (hunk 2 above)
Comment (kanika1894) [reply src/configure/helper/mustacheHelper.ts:195]: Using the sort function, we just have to return 1 or -1. More examples : https://medium.com/@pagalvin/sort-arrays-using-typescript-592fa6e77f1

**Pull request merged**